### DISCUSSION TITLE
Use @userstring if available instead of @clip

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#96] [Q&A] [answered]
`@text=((*@removelf(@text)*))
$text=((*$text*)) \n\n<a href="((*@pageRef*))">((*@clip*))</a>\n\n```\n\n((*@text*))\n\n```\n\n
`

This is a rule I use to snip text  in monospaced block format from webpages with html anchors.

Here I would like use `@userstring` when available instead of `@clip`
By defualt, `@clip` should be used. But if `@userstring` input is made, `@userstring` should be used instead of `@clip`
How to modify the rule for doing so.

**Comment by buggyj** [comment]
Try adding this before,

```
#empty=
#userstring=((*@exists(@userstring)*??*@userstring*??*#empty*))
@clip=((*!#userstring==#empty*??*#userstring*??*@abort()*))
```

Note that there is no text or spaces after `#empty=` above

**Reply by arunnbabu81** [reply]
> Try adding this before,
> 
> ```
> #empty=
> #userstring=((*@exists(@userstring)*??*@userstring*??*#empty*))
> @clip=((*!#userstring==#empty*??*#userstring*??*@abort()*))
> ```
> 
> Note that there is no text or spaces after `#empty=` above

This was working. I didn’t test the second one. Which one do you prefer to use?

**Reply by buggyj** [reply]
The second is easier to understand, they are both ok.

**Comment by buggyj** [comment, marked as answer]
Also this should work:

```
#empty=
#userstring=((*@exists(@userstring)*??*@userstring*??*#empty*))
@clip=((*!#userstring==#empty*??*#userstring*??*@clip*))
```
